server/bin/pbench-server-prep-shim-002.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+# This script is used to prepare the tarballs that a version 002 client
+# submits for further processing. It copies the tarballs and their MD5
+# sums to the archive (after checking) and sets the state links, so
+# that the dispatch script will pick them up and get the ball
+# rolling. IOW, it does impedance matching between version 002 clients
+# and the server scripts.
+
+import os
+import sys
+import glob
+import shutil
+import tempfile
+
+from pathlib import Path
+from pbench.server import PbenchServerConfig
+from pbench.common.exceptions import BadConfig
+from pbench.common.logger import get_pbench_logger
+from pbench.server.report import Report
+from pbench.server.utils import md5sum, quarantine
+
+
+_NAME_ = "pbench-server-prep-shim-002"
+
+
+class Results(object):
+    def __init__(
+        self, nstatus="", ntotal=0, ntbs=0, nquarantined=0, ndups=0, nerrs=0,
+    ):
+        self.nstatus = nstatus
+        self.ntotal = ntotal
+        self.ntbs = ntbs
+        self.nquarantined = nquarantined
+        self.ndups = ndups
+        self.nerrs = nerrs
+
+
+def fetch_config_val(config, errorfile):
+
+    qdir = config.get("pbench-server", "pbench-quarantine-dir")
+    if not qdir:
+        errorfile.write("Failed: getconf.py pbench-quarantine-dir pbench-server\n")
+        return None, None
+
+    qdir = Path(qdir).resolve()
+    if not qdir.is_dir():
+        errorfile.write(f"Failed: {qdir} does not exist, or is not a directory\n")
+        return None, None
+
+    # we are explicitly handling version-002 data in this shim
+    receive_dir_prefix = config.get("pbench-server", "pbench-receive-dir-prefix")
+    if not receive_dir_prefix:
+        errorfile.write("Failed: getconf.py pbench-receive-dir-prefix pbench-server\n")
+        return None, None
+
+    receive_dir = Path(f"{receive_dir_prefix}-002").resolve()
+    if not receive_dir.is_dir():
+        errorfile.write(
+            f"Failed: {receive_dir} does not exist, or is not a directory\n"
+        )
+        return None, None
+
+    return (qdir, receive_dir)
+
+
+def qdirs_check(qdir_val, qdir, logger):
+    try:
+        os.makedirs(qdir)
+    except FileExistsError:
+        # directory already exists, ignore
+        pass
+    except Exception:
+        logger.exception(
+            "os.mkdir: Unable to create {} destination directory: {}", qdir_val, qdir,
+        )
+        return None
+    return qdir
+
+
+def md5_check(tb, tbmd5, logger):
+    # read the md5sum from md5 file
+    try:
+        with tbmd5.open() as f:
+            archive_md5_hex_value = f.readline().split(" ")[0]
+    except Exception:
+        archive_md5_hex_value = None
+        logger.exception("Quarantine: Could not read {}", tbmd5)
+
+    # get hex value of the tarball's md5sum
+    try:
+        archive_tar_hex_value = md5sum(tb)
+    except Exception:
+        logger.exception("Quarantine: Could not read {}", tb)
```

Comment:
Same thing applies here, `archive_tar_hex_value` needs a value in order to avoid an another exception at line 97 below.